### PR TITLE
Mempool tweaks

### PR DESCRIPTION
**Pull request by someone235**
This PR:

* Increases mempool expiry time to 24h
* Increases orphan tx limit to 500
* Adds a compute-mass limit to the mempool (as a proxy to mempool memory size) and changing the eviction mechanism to prefer transactions with lower fee rate.

### CODE REVIEW

File: mining/src/mempool/validate_and_insert_transaction.rs [hunk 1]
```diff
@@ -73,9 +74,30 @@ impl Mempool {
...
-        })?;
+        let transaction_size = transaction.tx.estimate_mem_bytes();
+        let txs_to_remove = self.transaction_pool.limit_transaction_count(&transaction)?;
+        for x in txs_to_remove.iter() {
+            self.remove_transaction(x, true, TxRemovalReason::MakingRoom, format!(" for {}", transaction_id).as_str())?;
+            // self.transaction_pool.limit_transaction_count(&transaction) returns the
+            // smallest prefix of `ready_transactions` (sorted by ascending fee-rate)
+            // that makes enough room for `transaction`, but since each call to `self.remove_transaction`
+            // also removes all transactions dependant on `x` we might already have enough room.
+            if self.transaction_pool.len() < self.config.maximum_transaction_count
```
Comment (coderofstuff) [inline mining/src/mempool/validate_and_insert_transaction.rs:85]: Same question here. Shouldn't this be `self.transaction_pool.len() <= self.config.maximum_transaction_count` ?

File: mining/src/manager_tests.rs [hunk 2]
```diff
@@ -1116,6 +1117,47 @@ mod tests {
         // TODO: extend the test according to the golang scenario
     }
 
+    #[test]
+    fn test_evict() {
```
Comment (coderofstuff) [inline mining/src/manager_tests.rs:1121]: Please add comments to explain what's happening in this test. Maybe add inline comments.

From reading it, it creates 10 txs and a heavy tx and the expectation is that to accommodate the heavy tx, 6 of the lighter txs are evicted.

File: mining/src/mempool/model/transactions_pool.rs [hunk 3]
```diff
@@ -181,70 +187,49 @@ impl TransactionsPool {
...
-            } else {
-                let mut low_priority_txs = low_priority_txs.collect::<Vec<_>>();
-                if low_priority_txs.len() > trim_size {
-                    low_priority_txs.sort_by(|a, b| a.fee_rate().partial_cmp(&b.fee_rate()).unwrap());
-                    transactions_to_remove.extend_from_slice(&low_priority_txs[0..usize::min(trim_size, low_priority_txs.len())]);
-                } else {
-                    transactions_to_remove = low_priority_txs;
-                }
+        let transaction_size = transaction.tx.estimate_mem_bytes();
+        let mut txs_to_remove = Vec::with_capacity(1);
```
Comment (coderofstuff) [inline mining/src/mempool/model/transactions_pool.rs:202]: Why `Vec::with_capacity(1)`? Add a comment as to why. My guess is that the expected number of transactions to remove is almost always exactly `1`.

File: mining/src/mempool/model/transactions_pool.rs [hunk 4]
```diff
@@ -181,70 +187,49 @@ impl TransactionsPool {
...
     ///
-    /// An error is returned if the mempool is filled with high priority transactions.
-    pub(crate) fn limit_transaction_count(
-        &self,
-        free_slots: usize,
-        transaction: &MutableTransaction,
-    ) -> RuleResult<Vec<TransactionId>> {
-        assert!(free_slots > 0);
+    /// An error is returned if the mempool is filled with high priority transactions, or
+    /// at least one candidate for removal has a higher fee rate than `transaction`.
```
Comment (coderofstuff) [inline mining/src/mempool/model/transactions_pool.rs:196]:
```
at least one candidate for removal has a higher fee rate than `transaction`
```

is probably better stated as

```
there are not enough lower fee rate transactions that can be removed to accommodate `transaction`
```

**Review by coderofstuff — CHANGES_REQUESTED** [review]
Changes look sound, I have a few comments to be fixed.

**Review by coderofstuff — APPROVED** [review]

**Review by michaelsutton — APPROVED** [review]

**Pull request merged**